client/src/components/organisms/CoverImage/CoverImage.jsx

```diff
@@ -1,25 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './CoverImage.css';
+import { Button, Form } from 'react-bootstrap';
+import ImgDropAndCrop from '../../molecules/ImgDropAndCrop/ImgDropAndCrop';
 
-const Background =
-  'https://images.unsplash.com/photo-1540959733332-eab4deabeeaf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1542&q=80';
+const Background = 'https://images.unsplash.com/photo-1540959733332-eab4deabeeaf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1542&q=80';
 
 const CoverImage = () => {
+  const [timeline, setTimeline] = useState({
+    bg: Background, title: 'My Trip To Tokyo', tagline: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer posuere erat a ante.', author: 'Ayush Jaiswal',
+  });
+  const {
+    bg, title, tagline, author,
+  } = timeline;
+
+  const [edit, setEdit] = useState(false);
+
+  const handleClick = () => setEdit(!edit);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setEdit(!edit);
+  };
+  const handleChange = (e) => {
+    setTimeline({ ...timeline, [e.target.name]: e.target.value });
+  };
   return (
     <div
-      style={{ backgroundImage: `url(${Background})` }}
+      style={{ backgroundImage: `url(${bg})` }}
       className="cover-image"
     >
       <div className="layer">
         <div className="flex-container">
-          <h1>My Trip To Tokyo</h1>
+          <Button onClick={handleClick}>{edit ? 'Done' : 'Edit'}</Button>
+          {edit ? <ImgDropAndCrop /> : null}
+          {edit ? (
+            <Form onSubmit={handleSubmit}>
```

Comment:
What is triggering onSubmit event? Only enter key? Please add an action.

Reply:
Submit essentially triggers a change in the edit mode of the CoverImage. Once the context API has been finalised. It will trigger the change in the data of the timeline.

Reply:
> Submit essentially triggers a change in the edit mode of the CoverImage. Once the context API has been finalised. It will trigger the change in the data of the timeline.